forte/data/data_store.py

```diff
@@ -982,30 +998,101 @@ def co_iterator_annotation_like(
             yield entry
 
     def get(
-        self, type_name: str, include_sub_type: bool = True
+        self,
+        type_name: str,
+        include_sub_type: bool = True,
+        range_annotation: Optional[Tuple[int]] = None,
     ) -> Iterator[List]:
         r"""This function fetches entries from the data store of
-        type ``type_name``.
+        type ``type_name``. If `include_sub_type` is set to True and
+        ``type_name`` is in [Annotation, Group, List], this function also
+        fetches entries of subtype of ``type_name``. Otherwise, it only
+        fetches entries of type ``type_name``.
 
         Args:
             type_name: The fully qualified name of the entry.
             include_sub_type: A boolean to indicate whether get its subclass.
+            range_annotation: A tuple that contains the begin and end indices
+                of the searching range of entries.
 
         Returns:
             An iterator of the entries matching the provided arguments.
         """
+
+        def within_range(
+            entry: List[Any], range_annotation: Tuple[int]
+        ) -> bool:
+            """
+            A helper function for deciding whether an annotation entry is
+            inside the `range_annotation`.
+            """
+            if not self._is_annotation(entry[constants.ENTRY_TYPE_INDEX]):
+                return False
+            return (
+                entry[constants.BEGIN_INDEX]
+                >= range_annotation[constants.BEGIN_INDEX]
+                and entry[constants.END_INDEX]
+                <= range_annotation[constants.END_INDEX]
+            )
+
+        if type_name not in self.__elements:
+            raise ValueError(f"type {type_name} does not exist")
+        entry_class = get_class(type_name)
+        all_types = set()
         if include_sub_type:
-            entry_class = get_class(type_name)
-            all_types = []
-            # iterate all classes to find subclasses
             for type in self.__elements:
                 if issubclass(get_class(type), entry_class):
-                    all_types.append(type)
+                    all_types.add(type)
+        else:
+            all_types.add(type_name)
+        all_types = list(all_types)
+        all_types.sort()
```

Comment:
@mylibrar take a note that we will replace this with get subtype function later, including the sorting.

Reply:
Issue created here https://github.com/asyml/forte/issues/807